parser/src/parser/tests/mod.rs

```diff
@@ -1,18 +1,619 @@
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Index($idx),
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:ident [ $idx:literal ], $offset:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Index($idx),
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
+
+    ($name:literal [ $idx:literal ], $offset:literal) => {
+        ScalarExpr::SymbolAccess(SymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            ident!($name),
+            AccessType::Index($idx),
+            $offset,
+        ))
+    };
+
+    ($module:ident, $name:ident [ $idx:literal ], $offset:literal, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Resolved(ident!($module, $name)),
+            access_type: AccessType::Index($idx),
+            offset: $offset,
+            ty: Some($ty),
+        })
+    };
+}
+
+macro_rules! expr {
+    ($expr:expr) => {
+        Expr::try_from($expr).unwrap()
+    };
+}
+
+macro_rules! slice {
+    ($name:ident, $range:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Unresolved(NamespacedIdentifier::Binding(ident!($name))),
+            access_type: AccessType::Slice($range),
+            offset: 0,
+            ty: None,
+        })
+    };
+
+    ($name:ident, $range:expr, $ty:expr) => {
+        ScalarExpr::SymbolAccess(SymbolAccess {
+            span: miden_diagnostics::SourceSpan::UNKNOWN,
+            name: ResolvableIdentifier::Local(ident!($name)),
+            access_type: AccessType::Slice($range),
+            offset: 0,
+            ty: Some($ty),
+        })
+    };
+}
+
+macro_rules! bounded_access {
+    ($name:ident, $bound:expr) => {
+        ScalarExpr::BoundedSymbolAccess(BoundedSymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            SymbolAccess::new(miden_diagnostics::SourceSpan::UNKNOWN, ident!($name), AccessType::Default, 0),
+            $bound,
+        ))
+    };
+
+    ($name:ident, $bound:expr, $ty:expr) => {
+        ScalarExpr::BoundedSymbolAccess(BoundedSymbolAccess::new(
+            miden_diagnostics::SourceSpan::UNKNOWN,
+            SymbolAccess {
+                span: miden_diagnostics::SourceSpan::UNKNOWN,
+                name: ResolvableIdentifier::Local(ident!($name)),
+                access_type: AccessType::Default,
+                offset: 0,
+                ty: Some($ty),
```

Comment:
We should never be able to have a non-Felt type for a bounded symbol access (although maybe this is for testing invalid cases?)

Reply:
Yeah that can probably be hardcoded here. If it ever comes up that we need to specify arbitrary types for certain tests, we can handle that as an additional case.

Reply:
Ah, I know why this exists at the moment - for tests which do not run through semantic analysis, the AST nodes are untyped, so we need a case that is untyped, and one that is typed, even if that type is always the same. So I made `bounded_access` mirror `access` here.

Reply:
Makes sense. It might be clearer for the future to add a comment which says that